src/ui/zcl_abapgit_gui_page_repo_sett.clas.abap

```diff
@@ -242,11 +243,19 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_REPO_SETT IMPLEMENTATION.
     ASSERT sy-subrc = 0.
     lo_dot->set_starting_folder( ls_post_field-value ).
 
+    lt_sorted_fields = it_post_fields.
+    sort lt_sorted_fields.
+
+    LOOP AT lt_sorted_fields INTO ls_post_field WHERE name CP 'req_*'.
+      IF ls_post_field-name+4(3) = 'com'.
+        INSERT INITIAL LINE INTO TABLE lt_requirements ASSIGNING <ls_requirement>.
+        <ls_requirement>-component = ls_post_field-value.
+      ENDIF.
+    ENDLOOP.
```

Comment:
Well, I am sorry but I have absolutely no idea why this commit exists :(

Reply:
The previous code assumes it_post_fields comes in a certain order, in the java GUI with the JS fix comes down in a different one.
Might try to hack the form creation but instead I decided to copy and sort, then use 2 loops for creation of the table entries and setting the other fields.

I  don't particularly like this code but the alternatives I could think of were worse